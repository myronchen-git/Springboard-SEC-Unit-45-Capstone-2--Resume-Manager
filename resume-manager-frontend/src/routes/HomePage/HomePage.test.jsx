import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { describe, expect, it } from 'vitest';

import HomePage from './HomePage.jsx';

// ==================================================

describe('HomePage', () => {
  it('renders.', () => {
    render(
      <MemoryRouter>
        <HomePage />
      </MemoryRouter>
    );
  });

  it('matches snapshot.', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <HomePage />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('displays the correct text.', () => {
    // Act
    const { getByText } = render(
      <MemoryRouter>
        <HomePage />
      </MemoryRouter>
    );

    expect(getByText('Resume Manager')).toBeVisible();
    expect(getByText('Register', { exact: false })).toBeVisible();
    expect(getByText('Sign In', { exact: false })).toBeVisible();
  });
});
