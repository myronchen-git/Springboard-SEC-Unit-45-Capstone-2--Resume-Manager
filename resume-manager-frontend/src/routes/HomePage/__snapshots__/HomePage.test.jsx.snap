// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomePage > matches snapshot. 1`] = `
<DocumentFragment>
  <main
    class="HomePage"
  >
    <div
      class="HomePage-backdrop"
    >
      <h1
        class="HomePage-title"
      >
        Resume Manager
      </h1>
      <p
        class="HomePage-description"
      >
        Store all of your information in one resume and create tailored resumes from it that automatically update themselves.
      </p>
      <div
        class="HomePage-buttons"
      >
        <a
          data-discover="true"
          href="/register"
        >
          <button
            class="btn btn-primary"
          >
            Register
          </button>
        </a>
        <a
          data-discover="true"
          href="/signin"
        >
          <button
            class="btn btn-primary"
          >
            Sign In
          </button>
        </a>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
